File: Projeto Didio Pizza/Didio Pizza Node/backend/src/controller/pedidoController.js
import { Router } from 'express';
import * as pedidoService from '../service/pedidoService.js';
import { getAuthentication, onlyAdmin } from '../utils/jwt.js';

const endpoints = Router();
const auth = getAuthentication();
const adm = onlyAdmin();


endpoints.post('/pedido', auth, async (req, resp) => {
    try {
        let clienteId = req.user.id;
        let enderecoId = req.body.endereco_id;

        if (!enderecoId) {
            return resp.status(400).send({
                erro: 'Endereço é obrigatório.'
            });
        }

        let pedidoId = await pedidoService.fecharPedido(clienteId, enderecoId);

        resp.status(201).send({
            id: pedidoId
        });
    } catch (err) {
        resp.status(400).send({
            erro: err.message
        });
    }
});


endpoints.get('/pedido', auth, async (req, resp) => {
    try {
        let clienteId = req.user.id;

        let pedidos = await pedidoService.listarPedidos(clienteId);

        resp.status(200).send(pedidos);

    } catch (err) {
        resp.status(400).send({
            erro: err.message
        });
    }
});


endpoints.get('/pedido/:id', auth, async (req, resp) => {
    try {
        let clienteId = req.user.id;
        let pedidoId = Number(req.params.id);

        let pedido = await pedidoService.detalharPedido(clienteId, pedidoId);

        resp.status(200).send(pedido);

    } catch (err) {
        resp.status(404).send({
            erro: err.message
        });
    }
});


endpoints.put('/pedido/:id/status', auth, adm, async (req, resp) => {
    try {
        let pedidoId = Number(req.params.id);
        let status = req.body.status;

        if (!status) {
            return resp.status(400).send({
                erro: 'Status é obrigatório.'
            });
        }

        await pedidoService.atualizarStatusPedido(pedidoId, status);

        resp.status(204).send();

    } catch (err) {
        resp.status(400).send({
            erro: err.message
        });
    }
});


endpoints.get('/pedidos/admin', auth, adm, async (req, resp) => {
    try {
        let pedidos = await pedidoService.listarPedidosAdmin();

        resp.status(200).send(pedidos);

    } catch (err) {
        resp.status(400).send({
            erro: err.message
        });
    }
});



export default endpoints;

File: Projeto Didio Pizza/Didio Pizza Springboot/backend/src/main/java/com/pizzaria/didiopizza/controllers/PedidoController.java
package com.pizzaria.didiopizza.controllers;

import java.util.List;
import java.util.Map;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import com.pizzaria.didiopizza.security.JwtUtil;
import com.pizzaria.didiopizza.service.PedidoService;

@RestController
public class PedidoController {
    private final PedidoService service;
    private final JwtUtil jwt;

    public PedidoController(PedidoService s, JwtUtil j) {
        service = s;
        jwt = j;
    }

    @PostMapping("/pedido")
    public ResponseEntity<?> criar(
            @RequestHeader("Authorization") String a,
            @RequestBody Map<String, Object> b) {
        Object e = b.get("endereco_id");
        Integer enderecoId = e == null ? null : Integer.parseInt(String.valueOf(e));

        Object itens = b.get("itens");

        if (itens instanceof List<?> lista && !lista.isEmpty()) {
            @SuppressWarnings("unchecked")
            List<Map<String, Object>> itensPedido = (List<Map<String, Object>>) itens;

            return ResponseEntity.status(201)
                    .body(Map.of("id", service.fecharPedidoComItens(jwt.getId(a), enderecoId, itensPedido)));
        }

        return ResponseEntity.status(201)
                .body(Map.of("id", service.fecharPedido(jwt.getId(a), enderecoId)));
    }

    @GetMapping("/pedido")
    public List<Map<String, Object>> listar(
            @RequestHeader("Authorization") String a) {
        return service.listarPedidos(jwt.getId(a));
    }

    @GetMapping("/pedido/{id}")
    public Map<String, Object> detalhe(
            @RequestHeader("Authorization") String a,
            @PathVariable Integer id) {
        if ("ADMIN".equalsIgnoreCase(jwt.getTipo(a))) {
            return service.detalharPedidoAdmin(id);
        }

        return service.detalharPedido(jwt.getId(a), id);
    }

    @GetMapping("/pedido/{id}/status")
    public Map<String, Object> statusPublico(@PathVariable Integer id) {
        return service.buscarStatusPedido(id);
    }

    @PutMapping("/pedido/{id}/status")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void status(
            @RequestHeader("Authorization") String a,
            @PathVariable Integer id,
            @RequestBody Map<String, Object> b) {
        jwt.onlyAdmin(a);
        if (b.get("status") == null)
            throw new RuntimeException("Status é obrigatório.");
        service.atualizarStatusPedido(id, String.valueOf(b.get("status")));
    }

    @GetMapping("/pedidos/admin")
    public List<Map<String, Object>> admin(
            @RequestHeader("Authorization") String a) {
        jwt.onlyAdmin(a);
        return service.listarPedidosAdmin();
    }
}
